components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/SAMLSSOAuthenticator.java

```diff
@@ -294,6 +294,11 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
             stateInfoDO.setSpNameQualifier(spNameQualifier);
             context.setStateInfo(stateInfoDO);
 
+            //include the saml index with federated idp as a property map to the context
+            Map<String, Object> authenticatorProperties = new HashMap<>();
+            context.setProperties(authenticatorProperties);
```

Comment:
Why do you need to call context.setProperties() with an empty map? is the properties varible is null?

Reply:
If that is the case, why not context.setProperties(new HashMap<String, Object>())?